core/src/main/scala/org/http4s/util/CaseInsensitiveString.scala

```diff
@@ -16,14 +34,61 @@ sealed class CaseInsensitiveString private (val value: String) extends CharSeque
 
   override def toString: String = value
 
-  def length(): Int = value.length
+  def apply(n: Int): Char =
+    toString.charAt(n)
+  def charAt(n: Int): Char =
+    apply(n)
```

Comment:
I think you're going to box then unbox here since `apply(Int)` comes from a generic trait.